specification/synapse/data-plane/Microsoft.Synapse/preview/2021-07-01-preview/artifacts.json

```diff
@@ -0,0 +1,5064 @@
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/NotebookName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "request",
+            "description": "proposed new name.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ArtifactRenameRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "202": {
+            "description": "Accepted."
+          },
+          "default": {
+            "description": "An error response received from the Azure Synapse service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/kqlScripts": {
+      "get": {
+        "tags": [
+          "kqlScripts"
+        ],
+        "operationId": "KqlScripts_GetAll",
+        "description": "Get all KQL scripts",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/KqlScripts_ResourceCollectionResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#/definitions/ErrorContract"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "KqlScripts_GetAll": {
+            "$ref": "./examples/KqlScripts_GetAll.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/kqlScripts/{kqlScriptName}": {
+      "put": {
+        "tags": [
+          "kqlScripts"
+        ],
+        "operationId": "KqlScript_CreateOrUpdate",
+        "description": "Creates or updates a KQL Script",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KqlScriptName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "kqlScript",
+            "description": "KQL script",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KqlScriptResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/KqlScriptResource"
+            }
+          },
```

Comment:
An operation that returns a 202 should not return any other success status code.

Reply:
But this is a long running operation and eventually it can returns 200 when using in LRO way

Reply:
@idear1203 -- We're in the process of refining the guidance on long running operations. You can read about them in the [updated guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations--jobs). 
Also, @mikekistler has a doc that we are working on as well focusing on LROs that we'll eventually merge into the guidelines. Here's a [link](https://github.com/mkistler/api-guidelines/blob/lro/azure/LongRunningOperations.md) if you are interested.